### ISSUE TITLE
HTTP_VERBS causing requests to be triggered during Vue's reactivity walk

### ISSUE THREAD

**Issue by SpyTec** [not-an-aardvark/snoowrap#177]
When an object is added to a VueJS instance in some manner, Vue changes all the fields to be getters and setters to be able to detect changes. One caveat of this is that if a snoowrap instance or snoowrap object is added it always sends off a request to Reddit implicitly.

This seems to be due to `HTTP_VERBS`. Could this be refactored?

https://github.com/not-an-aardvark/snoowrap/blob/443583c97b8754c559112ee5fe4abfa8c46ad8cc/src/objects/RedditContent.js#L125-L129

**Comment by not-an-aardvark** [comment]
For legacy reasons, snoowrap objects have getters with side-effects that cause requests to be sent to reddit. To disable this behavior you can use

```js
const r = new snoowrap({ /* ... */ });
r.config({ proxies: false });
```

**Comment by SpyTec** [comment]
Even with `proxies: false` in the config I'm still getting the request side-effect when adding an object or snoowrap instance in VueJS.

**Comment by not-an-aardvark** [comment]
Are you sure you're setting the `proxies: false` option before passing the object to Vue or making any requests?  One of the lines in the stacktrace is [here](https://github.com/not-an-aardvark/snoowrap/blob/884d6ee0ba07a77e9acb2635db9e3132f1a0e9e5/src/objects/RedditContent.js#L23), which should only get reached if proxies are enabled when an object from reddit is fetched.

**Comment by SpyTec** [comment]
Yes. Currently I set snoowrap in a singleton which calls `reddit.config({ proxies: false });`. Then in a Vue component I accessed moderated subreddits as such:

```javascript
console.log("proxies", this.reddit.config()['proxies']);
this.subredditList = await this.reddit.getModeratedSubreddits();
```

Proxies prints `false` and the moderated subreddits return, but with the sideeffect of the request going off once the Listing is added to the Vue component's data object. ([Different stacktrace for the component than the image above](https://i.imgur.com/7oVnncA.png))

I can try to create a simple VueJS project this weekend if you want to take a look closer
